src/message.js

```diff
@@ -304,17 +304,18 @@ export class Message {
 
   /**
    * Encrypt the message either with public keys, passwords, or both at once.
-   * @param  {Array<Key>} keys           (optional) public key(s) for message encryption
-   * @param  {Array<String>} passwords   (optional) password(s) for message encryption
-   * @param  {Object} sessionKey         (optional) session key in the form: { data:Uint8Array, algorithm:String, [aeadAlgorithm:String] }
-   * @param  {Boolean} wildcard          (optional) use a key ID of 0 instead of the public key IDs
-   * @param  {Date} date                 (optional) override the creation date of the literal package
-   * @param  {Array<Object>} userIds     (optional) user IDs to encrypt for, e.g. [{ name:'Robert Receiver', email:'robert@openpgp.org' }]
-   * @param  {Boolean} streaming         (optional) whether to process data as a stream
+   * @param  {Array<Key>} keys                            (optional) public key(s) for message encryption
+   * @param  {Array<String>} passwords                    (optional) password(s) for message encryption
+   * @param  {Object} sessionKey                          (optional) session key in the form: { data:Uint8Array, algorithm:String, [aeadAlgorithm:String] }
+   * @param  {Boolean} wildcard                           (optional) use a key ID of 0 instead of the public key IDs
+   * @param  {Date} date                                  (optional) override the creation date of the literal package
+   * @param  {Array<Object>} userIds                      (optional) user IDs to encrypt for, e.g. [{ name:'Robert Receiver', email:'robert@openpgp.org' }]
+   * @param  {Boolean} streaming                          (optional) whether to process data as a stream
+   * @param  {Array<module:type/keyid>} encryptionKeyIds  (optional) array of key IDs to use for encryption. Each encryptionKeyIds[i] corresponds to publicKeys[i]
    * @returns {Promise<Message>}                   new message with encrypted content
    * @async
    */
-  async encrypt(keys, passwords, sessionKey, wildcard = false, date = new Date(), userIds = [], streaming) {
+  async encrypt(keys, passwords, sessionKey, wildcard = false, date = new Date(), userIds = [], streaming, encryptionKeyIds = []) {
```

Comment:
This is very nitpicky, but could you make the order:
```suggestion
  async encrypt(keys, passwords, sessionKey, wildcard = false, encryptionKeyIds = [], date = new Date(), userIds = [], streaming) {
```
(for symmetry with the existing `getEncryptionKey` function, which takes `keyId, date, userId`)?

(And similarly in the other functions.)